mmcls/models/backbones/mlp_mixer.py

```diff
@@ -0,0 +1,198 @@
+import torch.nn as nn
+from mmcv.cnn import build_activation_layer, build_norm_layer
+from mmcv.cnn.bricks.drop import build_dropout
+from mmcv.runner.base_module import BaseModule, Sequential
+from mmcv.utils.misc import to_2tuple
+
+from mmcls.models.utils import PatchEmbed
+from ..builder import BACKBONES
+
+
+class MixerBlock(nn.Module):
+    """MlpMixer block.
+
+    Basic module of `MLP-Mixer: An all-MLP Architecture for Vision
+    <https://arxiv.org/pdf/2105.01601.pdf>`_
+
+    Some code is borrowed and modified from timm.
+
+    Args:
+        dim (int): Embedding dimension.
+        seq_len (int): Length of the sequence, mally equivalent to um patches.
+        mlp_ratio (tuple | int): Multiplier for the D_s and D_c. Default
+            to (0.5, 4.0)
+        norm_cfg (dict): Configuration for the normalization layer. Default to
+            dict(type='LN', eps=1e-6).
+        act_layer_cfg (dict): Configuration for the activation layer. Default
+            to dict(type='GELU').
+        drop_rate (float): Dropout rate for the dropout layer in MixerBlock.
+            Default to 0.
+        drop_path_rate (float | list): Drop path rate for the DropPath (
+            Stochastic Depth) in Mixer. Default to 0.
+    """
+
+    def __init__(self,
+                 dim,
+                 seq_len,
+                 mlp_ratio=(0.5, 4.0),
+                 norm_cfg=dict(type='LN', eps=1e-6),
+                 act_layer_cfg=dict(type='GELU'),
+                 drop_rate=0.,
+                 drop_path_rate=0.):
+        super().__init__()
+        tokens_dim, channels_dim = [int(x * dim) for x in to_2tuple(mlp_ratio)]
+        self.norm1 = build_norm_layer(norm_cfg, dim)[1]
+        self.mlp_tokens = self.__build_mlp(
+            seq_len,
+            tokens_dim,
+            act_layer_cfg=act_layer_cfg,
+            drop_rate=drop_rate)
+        self.drop_path = build_dropout(
+            dict(type='DropPath', drop_prob=drop_path_rate)
+        ) if drop_path_rate > 0. else nn.Identity()
```

Comment:
Well, the purpose of `build_dropout` is to specify the type of dropout by the config dict. If we just use `DropPath` here, instantiating a `DropPath` is enough.